integration/connection_string_builder.h

```diff
@@ -166,6 +166,38 @@ class ConnectionStringBuilder {
     return *this;
   }
 
+  ConnectionStringBuilder& withEnableCustomEndpointMonitoring(const bool& enable_custom_endpoint_monitoring) {
+    length +=
+        sprintf(conn_in + length, "ENABLE_CUSTOM_ENDPOINT_MONITORING=%d;", enable_custom_endpoint_monitoring ? 1 : 0);
+    return *this;
+  }
+  ConnectionStringBuilder& withCustomEndpointRegion(const std::string& region) {
+    length += sprintf(conn_in + length, "CUSTOM_ENDPOINT_REGION=%s;", region.c_str());
+    return *this;
+  }
+
+  ConnectionStringBuilder& withShouldWaitForInfo(const bool& should_wait_for_info) {
+    length += sprintf(conn_in + length, "WAIT_FOR_CUSTOM_ENDPOINT_INFO=%d;", should_wait_for_info ? 1 : 0);
+    return *this;
+  }
+
+  ConnectionStringBuilder& withCustomEndpointInfoRefreshRateMs(const long& custom_endpoint_info_refresh_rate_ms) {
+    length +=
+        sprintf(conn_in + length, "CUSTOM_ENDPOINT_INFO_REFRESH_RATE_MS=%d;", custom_endpoint_info_refresh_rate_ms);
+    return *this;
```

Comment:
No real impact since default is already a `long` but just to keep consistent with your other 2 lines
```suggestion
    length +=
        sprintf(conn_in + length, "CUSTOM_ENDPOINT_INFO_REFRESH_RATE_MS=%ld;", custom_endpoint_info_refresh_rate_ms);
    return *this;
```